src/split_packets.py

```diff
@@ -0,0 +1,365 @@
+
+            # check the packet shopuld not be filtered out
+            if mag_only:
+                if apid < APID_MAG_START or apid > APID_MAG_END:
+                    continue
+
+            if apid_filter:
+                if apid not in apid_filter:
+                    continue
+
+            is_science = False
+            if apid == 0x41C or apid == 0x42C:
+                is_science = True
+
+                fileLikeObject.seek(0)
+                pkt = sciPktDefinition.load(fileLikeObject, include_primary_header=True)
+
+            # Save the single packet to it's own .bin file?
+            if not summarise_only:
+                newFileName = (
+                    packet_file.parent
+                    / str(apid)
+                    / f"{pkt['SHCOARSE'][0]}-{pkt['CCSDS_SEQUENCE_COUNT'][0]}.bin"
+                )
+
+                if newFileName.exists():
+                    print(f"Existing packet found: {newFileName} - skipped it")
+                    exit_code = 1
+                else:
+                    if not newFileName.parent.exists():
+                        newFileName.parent.mkdir(parents=True)
+
+                    with open(newFileName, "wb") as f:
+                        f.write(packet_bytes)
+
+            packet_counter += 1
+
+            if not no_report:
+                report_file.write(
+                    f"{apid},{pkt['CCSDS_SEQUENCE_COUNT'][0]},{pkt['CCSDS_PACKET_LENGTH'][0]},{pkt['SHCOARSE'][0]}\n"
+                )
+                if is_science:
+                    sci_report_file.write(
+                        f"{apid},{pkt['CCSDS_SEQUENCE_COUNT'][0]},{pkt['CCSDS_PACKET_LENGTH'][0]},{pkt['SHCOARSE'][0]},"
+                        + f"{pkt['PUS_SSUBTYPE'][0]},"
+                        + f"{pkt['COMPRESSION'][0]},"
+                        + f"{pkt['FOB_ACT'][0]},"
+                        + f"{pkt['FIB_ACT'][0]},"
+                        + f"{pkt['PRI_SENS'][0]},"
+                        + f"{pkt['PRI_VECSEC'][0]},"
+                        + f"{pkt['SEC_VECSEC'][0]},"
+                        + f"{pkt['PRI_COARSETM'][0]},"
+                        + f"{pkt['PRI_FNTM'][0]},"
+                        + f"{pkt['SEC_COARSETM'][0]},"
+                        + f"{pkt['SEC_FNTM'][0]}"
+                        + "\n"
+                    )
+
+            if limit > 0 and packet_counter >= limit:
+                print(f"Limit of {limit} packets reached")
+                break
+
+    if not summarise_only:
+        print(
+            f"Saved {packet_counter} packets from {packet_file} to {packet_file.parent} ({size} bytes)"
+        )
+
+
+def process_multi_file(
+    globPath, ctx, report_file_path, no_report, summarise_only, limit, apids, mag_only
+):
+    multifile_exit_code = 0
+    files = 0
+    global is_multi_file
+    is_multi_file = True
+    for filename in glob.glob(globPath):
+        files += 1
+        try:
+            result = ctx.invoke(
```

Comment:
Why `invoke` here rather than just calling the `process_single_file` function directly?